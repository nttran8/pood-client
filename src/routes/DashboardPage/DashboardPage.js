// Libraries
import React, { Component } from "react";

// Components
import Dashboard from "../../components/Dashboard/Dashboard";

// Data
import PoodContext from "../../contexts/PoodContext";


export default class DashboardPage extends Component {
  // Set context for class
  static contextType = PoodContext;

  componentDidMount() {
    this.context.clearError();
  }

  render() {
    console.log(this.context.user);
    const { error } = this.context;
    return (
      <section className="DashboardPage">
        {error ? (
          <p className="red">Poop. There was an error!</p>
        ) : (
          <Dashboard />
        )}
      </section>
    );
  }
}
