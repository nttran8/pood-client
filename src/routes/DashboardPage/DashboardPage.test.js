import React from "react";
import TestRenderer from "react-test-renderer";
import DashboardPage from "./DashboardPage";
import { BrowserRouter as Router } from "react-router-dom";

it(`Renders DashboardPage UI as expected`, () => {
  const tree = TestRenderer.create(
    <Router>
      <DashboardPage />
    </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
