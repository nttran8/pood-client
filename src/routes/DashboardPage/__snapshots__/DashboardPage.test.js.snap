// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders DashboardPage UI as expected 1`] = `
<section
  className="DashboardPage"
>
  <div
    className="Dashboard"
  >
    <div
      className="DashboardPage__left"
    >
      <a
        href="/log"
        onClick={[Function]}
      >
        <button
          className="button logButton"
        >
          Add Log
        </button>
      </a>
      <ul
        onClick={[Function]}
      >
        <p
          className="emptyMessage"
        >
          Your list is empty. Dung forget to log!
        </p>
      </ul>
    </div>
    <div
      className="DashboardPage__right"
    />
  </div>
</section>
`;
